src/gossip/prune.zig

```diff
@@ -0,0 +1,208 @@
+            .wallclock = now,
+        };
+    }
+
+    pub fn deinit(self: Self, allocator: std.mem.Allocator) void {
+        allocator.free(self.prunes);
+    }
+
+    const PruneSignableData = struct {
+        pubkey: Pubkey,
+        prunes: []const Pubkey,
+        destination: Pubkey,
+        wallclock: u64,
+    };
+
+    const PruneSignableDataWithPrefix = struct {
+        prefix: []const u8 = PRUNE_DATA_PREFIX,
+        pubkey: Pubkey,
+        prunes: []const Pubkey,
+        destination: Pubkey,
+        wallclock: u64,
+    };
+
+    pub fn initRandom(random: std.rand.Random, keypair: *const KeyPair) !PruneData {
+        var self = PruneData{
+            .pubkey = Pubkey.fromPublicKey(&keypair.public_key),
+            .prunes = &[0]Pubkey{},
+            .signature = Signature.init(.{0} ** 64),
+            .destination = Pubkey.initRandom(random),
+            .wallclock = getWallclockMs(),
+        };
+        try self.sign(keypair);
+
+        return self;
+    }
+
+    pub fn sign(self: *PruneData, keypair: *const KeyPair) !void {
+        try self.signWithoutPrefix(keypair);
+    }
+
+    pub fn signWithoutPrefix(self: *PruneData, keypair: *const KeyPair) !void {
+        var slice: [PACKET_DATA_SIZE]u8 = undefined;
+        const signable_data = PruneSignableData{
+            .pubkey = self.pubkey,
+            .prunes = self.prunes,
+            .destination = self.destination,
+            .wallclock = self.wallclock,
+        };
+        const out = try bincode.writeToSlice(&slice, signable_data, bincode.Params{});
+        var signature = try keypair.sign(out, null);
+        self.signature.data = signature.toBytes();
+    }
+
+    pub fn signWithPrefix(self: *PruneData, keypair: *const KeyPair) !void {
+        var slice: [PACKET_DATA_SIZE]u8 = undefined;
+        const signable_data = PruneSignableDataWithPrefix{
+            .pubkey = self.pubkey,
+            .prunes = self.prunes,
+            .destination = self.destination,
+            .wallclock = self.wallclock,
+        };
+        const out = try bincode.writeToSlice(&slice, signable_data, bincode.Params{});
+        var signature = try keypair.sign(out, null);
+        self.signature.data = signature.toBytes();
+    }
+
+    pub fn verify(self: *const PruneData) !void {
+        self.verifyWithoutPrefix() catch |err| switch (err) {
+            error.InvalidSignature => try self.verifyWithPrefix(),
+            else => return err,
+        };
+    }
+
+    pub fn verifyWithoutPrefix(self: *const PruneData) !void {
+        var slice: [PACKET_DATA_SIZE]u8 = undefined;
+        const signable_data = PruneSignableData{
+            .pubkey = self.pubkey,
+            .prunes = self.prunes,
+            .destination = self.destination,
+            .wallclock = self.wallclock,
+        };
+        const out = try bincode.writeToSlice(&slice, signable_data, bincode.Params{});
+        if (!try self.signature.verify(self.pubkey, out)) {
```

Comment:
The following is within 100 col limit, however I understand you may favour the clarity of a block in this context so feel free to disregard
```suggestion
        if (!try self.signature.verify(self.pubkey, out)) return error.InvalidSignature;
```

Reply:
usually would be fine with this but since the return is a key part of the method would rather keep it indented to make it stand out more